src/python/pants/rules/core/test.py

```diff
@@ -76,6 +80,30 @@ def non_member_error_message(subject):
     return None
 
 
+@dataclass(frozen=True)
+class AddressAndTestResult:
+  address: Address
+  test_result: Optional[TestResult]  # If None, target was not a test target.
+
+  @staticmethod
+  def is_testable(
+    adaptor_with_origin: TargetAdaptorWithOrigin, *, union_membership: UnionMembership,
```

Comment:
We no longer need to use `address_origin_map` because the `TargetAdaptorWithOrigin` has all the information we need about the `OriginSpec`.

In a followup, we likely want to remove `address_origin_map`. Generally, the rule could instead request an `AddressWithOrigin` if they care about what the original spec was.

Reply:
Reasonable (with the caveat that we probably want to project an `Address` and an `Origin` as two separate params: see above.)